vulnerabilities/pipelines/exploitdb.py

```diff
@@ -0,0 +1,95 @@
+import csv
+import io
+import logging
+
+import requests
+
+from vulnerabilities.models import Alias
+from vulnerabilities.models import Exploit
+from vulnerabilities.models import VulnerabilityReference
+from vulnerabilities.models import VulnerabilityRelatedReference
+from vulnerabilities.pipelines import VulnerableCodePipeline
+
+logger = logging.getLogger(__name__)
+
+
+class ExploitDBImproverPipeline(VulnerableCodePipeline):
+    """
+    ExploitDB Improver Pipeline: Fetch ExploitDB data, iterate over it to find the vulnerability with
+    the specified alias, and create or update the ref and ref-type accordingly.
+    """
+
+    exploit_data = None
+
+    license_expression = "GPL-2.0"
+
+    @classmethod
+    def steps(cls):
+        return (
+            cls.fetch_exploits,
+            cls.add_exploit,
+        )
+
+    def fetch_exploits(self):
+        exploit_db_url = (
+            "https://gitlab.com/exploit-database/exploitdb/-/raw/main/files_exploits.csv"
+        )
+        response = requests.get(exploit_db_url)
+        self.exploit_data = io.StringIO(response.text)
+
+    def add_exploit(self):
+        csvreader = csv.reader(self.exploit_data)
+
+        header = next(csvreader)
+        for row in csvreader:
+
+            aliases = row[11].split(";")
+
+            for raw_alias in aliases:
+
+                alias = Alias.objects.get_or_none(alias=raw_alias)
+                if not alias:
+                    continue
+
+                vul = alias.vulnerability
```

Comment:
If more than one alias points to the same vulnerability, then we should not repeat the below process each time. We should simply skip the already processed vulnerability.

Reply:
@keshav-space After conducting some investigations, I believe we may have an exploit for multiple vulnerabilities That is why I think we shouldn't skip this and we should store vulnerabilities as a set and iterate over it. 


Example 1 ( Exploit-DB ):
```
44806,exploits/linux/local/44806.txt,"Procps-ng - Multiple Vulnerabilities",2018-05-30,"Qualys Corporation",local,linux,,2018-05-30,2018-05-30,0,CVE-2018-1124;CVE-2018-1123;CVE-2018-1122;CVE-2018-1121;CVE-2018-1120,Local,,,,
```

Example 2 ( Metasploit ):
```
"post_windows/gather/credentials/whatsupgold_credential_dump": {
    "name": "WhatsUp Gold Credentials Dump",
    "fullname": "post/windows/gather/credentials/whatsupgold_credential_dump",
    "aliases": [

    ],
    "rank": 0,
    "disclosure_date": "2022-11-22",
    "type": "post",
    "author": [
      "sshah <sshah@assetnote.io>",
      "npm <npm@cesium137.io>"
    ],
    "description": "This module exports and decrypts credentials from WhatsUp Gold to a CSV file;\n          it is intended as a post-exploitation module for Windows hosts with WhatsUp\n          Gold installed. The module has been tested on and can successfully decrypt\n          credentials from WhatsUp versions 11.0 to the latest (22.x). Extracted\n          credentials are automatically added to loot.",
    "references": [
      "CVE-2022-29845",
      "CVE-2022-29846",
      "CVE-2022-29847",
      "CVE-2022-29848",
      "URL-https://nvd.nist.gov/vuln/detail/CVE-2022-29845",
      "URL-https://nvd.nist.gov/vuln/detail/CVE-2022-29846",
      "URL-https://nvd.nist.gov/vuln/detail/CVE-2022-29847",
      "URL-https://nvd.nist.gov/vuln/detail/CVE-2022-29848",
      "URL-https://blog.assetnote.io/2022/06/09/whatsup-gold-exploit/"
    ],
  ...
```

Reply:
That makes sense, thanks for the clarification!